eng/scripts/Language-Settings.ps1

```diff
@@ -8,9 +8,31 @@ $BlobStorageUrl = "https://azuresdkdocs.blob.core.windows.net/%24web?restype=con
 
 function Get-dotnet-PackageInfoFromRepo ($pkgPath, $serviceDirectory, $pkgName)
 {
-  $projectPath = Join-Path $pkgPath "src" "$pkgName.csproj"
-  if (Test-Path $projectPath)
+  $projDirPath = (Join-Path $pkgPath "src")
+
+  if (!(Test-Path $projDirPath))
+  {
+    return $null
+  }
+
+  if ($pkgName)
   {
+    $projectPath = Join-Path $projDirPath "$pkgName.csproj"
+  }
+  else
+  {
+    $projectPath = (Resolve-Path (Join-Path $projDirPath "*.csproj")).path
+  }
+
+  if ($projectPath.Count -gt 1)
```

Comment:
This will fail once we turn on strict mode because you are essentially changing types. I suggest using a different variable name and doing this inside the else block above.